src/Pages/EmailList/EmailList.js

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import './email-list.css';
+import { getAllUsers } from '../../APIFunctions/User';
+import {
+  Button,
+  ButtonDropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem
+} from 'reactstrap';
+
+export default class EmailList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      users: [],
+      filteredUsers: [],
+      filtered: false,
+      dropdownOpen: false
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.user) {
+      this.getUsers();
+    }
+  }
+
+  handleToggle = () => {
+    this.setState({ dropdownOpen: !this.state.dropdownOpen });
+  }
+
+  getUsers = async () => {
+    const apiResponse = await getAllUsers(this.props.user.token);
+    if (!apiResponse.error){
+      let userData = apiResponse.responseData;
+      let users = [];
+      for(let i=0; i<userData.length; i++){
+        users.push({
+          email: userData[i].email,
+          accessLevel: userData[i].accessLevel,
+          membershipValidUntil: userData[i].membershipValidUntil,
+          emailOptIn: userData[i].emailOptIn
+        });
+      }
+      this.setState({users});
+    }
+  }
+
+  /*
+  Filters users via filterID parameter
+    0: Non-Valid Members
+    1: Members
+    2: Officers
+    3: Admin
+    4: Everyone Valid
+  */
+  filterUsers = (filterID) => {
+    let filteredUsers = this.state.users.filter((user) => {
+      const d = new Date();
+      if(filterID === 4){ // Everyone valid
+        if(user.accessLevel>=1 && user.accessLevel<=4 &&
+               user.membershipValidUntil > d.toISOString()){
+          return user;
+        }
+      } else if(filterID === 0){ // Non-valid/Pending
+        if(user.accessLevel === 0 || user.accessLevel === -1){
+          return user;
+        }
+      } else{ // Member/Officer/Executive
+        if(user.accessLevel === filterID  &&
+               user.membershipValidUntil > d.toISOString()){
+          return user;
+        }
+      }
+      return '';
+    });
+    this.setState({ filteredUsers, filtered: true });
+  }
+
+  // Copies all emails onto clipboard
+  handleCopyEmails = () => {
+    let range = document.createRange();
+    range.selectNode(document.getElementById('emails'));
+    window.getSelection().removeAllRanges(); // clear current selection
+    window.getSelection().addRange(range); // to select text
+    document.execCommand('copy');
+    window.getSelection().removeAllRanges();// to deselect
+    document.getElementById('copy-notification').style.display = 'block';
+  }
+
+  render() {
+    let filterItems = [
```

Comment:
you can use the enums and function to convert the enum values [here](https://github.com/SCE-Development/Core-v4/blob/addd3e75b606aeefca6cfef47d7d4d56dc6810b8/src/Enums.js#L52)

Reply:
why do we still have this

Reply:
The labels in the Enum file are 'VALID', 'NON_VALID', and 'ALL'. It doesn't look that good on the frontend dropdown, so I created the labels 'Valid Members', 'Non-Valid Members' and 'Everyone'